pandaserver/dataservice/closer.py

```diff
@@ -0,0 +1,400 @@
+"""
+update dataset DB, and then close dataset and start Activator if needed
+
+"""
+
+import re
+import sys
+from typing import Dict, List
+
+from pandacommon.pandalogger.PandaLogger import PandaLogger
+
+from pandaserver.config import panda_config
+from pandaserver.dataservice import Notifier
+from pandaserver.dataservice.Activator import Activator
+from pandaserver.taskbuffer import EventServiceUtils
+
+# logger
+_logger = PandaLogger().getLogger("closer")
+
+
+def init_logger(p_logger: PandaLogger) -> None:
+    """
+    Redirect logging to parent as it doesn't work in nested threads
+
+    Args:
+        p_logger (PandaLogger): The parent logger.
+    """
+    # redirect logging to parent as it doesn't work in nested threads
+    global _logger
+    _logger = p_logger
+    Notifier.initLogger(_logger)
+
+
+class Closer:
+    """
+    Update dataset DB, and then close dataset and start Activator if needed
+    """
+
+    # constructor
+    def __init__(self, taskBuffer, destination_db_blocks: List[str], job, panda_ddm: bool = False,
+                 dataset_map: Dict = {}) -> None:
+        """
+        Constructor
+
+        Args:
+            task_buffer: Task buffer.
+            destination_db_blocks (List[str]): Destination DB blocks.
+            job: Job.
+            panda_ddm (bool, optional): Panda DDM flag. Defaults to False.
+            dataset_map (Dict, optional): Dataset map. Defaults to {}.
+        """
+        self.task_buffer = taskBuffer
+        self.destination_db_blocks = destination_db_blocks
```

Comment:
what's a destination DB block? I think DB=dispatch block. IMO it would help to rename this variable.